### PR TITLE
[setups/osflow] update README

### PR DESCRIPTION
**Pull request by stnolting**
@umarcor @lab-mathias-claussen 

I have started to (finally) update the osflow README.
The first step is to add a tutorial-like sections how to add new FPGAs/board (idea coming from #218).
The other sections also require a makeover.

Help would be highly appreciated! :wink:

### CODE REVIEW

**Comment by stnolting** [comment]
I have finished the step-by-step guide. I am not sure if this is really helpful, because it just wraps up the commits in #217. What do you think (@umarcor @lab-mathias-claussen )?

**Comment by stnolting** [comment]
I think this is "good enough" for a first version. We can improve this later. :wink:

File: setups/osflow/README.md [hunk 1]
```diff
@@ -1,27 +1,38 @@
 # Exemplary FPAG Board Setups - Using Open Source Toolchains
 
+* [Folder Structure](#Folder-Structure)
```
Comment (umarcor) [inline setups/osflow/README.md:3]: GitHub and other services do show a table of contents automatically. Not sure about being worth maintaining it manually...

File: setups/osflow/README.md [hunk 2]
```diff
@@ -1,27 +1,38 @@
...
+* `boards`: board-specific _partial makefiles_ (used by main makefile "`Makefile`") for generating bitstreams
 * `board_top`: board-specific top entities (board wrappers; may include FPGA-specific modules)
 * `constraints`: physical constraints (mainly pin mappings)
 * `devices`: FPGA-specific primitives and optimized processor modules (like memories)
 
 
 ## Prerequisites
 
-:construction: **TODO - Under Construction** :construction:
+:construction: TODO :construction:
```
Comment (umarcor) [inline setups/osflow/README.md:26]: I will update this. Shall I wait to do it in neorv32-setups? #254.

**Review by umarcor — COMMENTED** [review]
Overall, very good guidelines!

File: setups/osflow/README.md (hunk 1 above)
Comment (stnolting) [reply setups/osflow/README.md:3]: Good point. We can rework that after the "spin-off" from #254

File: setups/osflow/README.md (hunk 2 above)
Comment (stnolting) [reply setups/osflow/README.md:26]: > Shall I wait to do it in neorv32-setups? #254.

:+1:

File: setups/osflow/README.md (hunk 1 above)
Comment (umarcor) [reply setups/osflow/README.md:3]: Addressed in stnolting/neorv32-setups#4.

File: setups/osflow/README.md (hunk 2 above)
Comment (umarcor) [reply setups/osflow/README.md:26]: Addressed in stnolting/neorv32-setups#4.

**Pull request merged**